### ISSUE TITLE
Non-functioning autocomplete/navigation in large codebase

### ISSUE THREAD

**Issue by Louisdepace** [bscan/PerlNavigator#13]
I am attempting to run the extension in an environment with a fairly large perl codebase. My initial attempt showed navigation and autocomplete to be non-functional. The extension output log reported an error running modhunter (pasted below). The error appears to be thrown from the try loop on line 98 of server/src/navigation.ts. I tried experimenting with the buffer size passed to execFile on line 100 of the same file. Expanding it by a factor of 4 (though not 2) seemed to resolve the error and restore functionality.

`        const out = await async_execFile(settings.perlPath, perlParams, {timeout: 90000, maxBuffer: 3 * 1024 * 1024 * 4});
`

I wanted to know if there was a compelling reason for the buffer size here, or if the size was chosen somewhat arbitrarily. If the latter, is this something that could be expanded or made configurable so as to work on our large codebase?

**Error message:**
ModHunter failed. You will lose autocomplete on importing modules. Not a huge deal
RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
    at new NodeError (node:internal/errors:371:5)
    at Socket.onChildStdout (node:child_process:461:14)
    at Socket.emit (node:events:526:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:285:11)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23) {
  code: 'ERR_CHILD_PROCESS_STDIO_MAXBUFFER',

**Comment by bscan** [comment]
Thank you for the very detailed report! That value was chosen mostly arbitrarily.  The node.js library has 1024*1024 (1 MB) as the maxBuffer default value: https://nodejs.org/api/child_process.html#child_processexecfilefile-args-options-callback 

I increased to 3 MB as necessary during my initial testing and never revisited it. However, I also use execFile for running perl -c and dumping the symbol table, and it uses 20 MB as the maxBuffer value (since there may be quite a few symbols found) and have never had an issue with this higher value.
 https://github.com/bscan/PerlNavigator/blob/ff6f94f015ff640551c8ee72a6bef992b3d79038/server/src/diagnostics.ts#L34 

I just increased the value to 20 MB for consistency (although even that choice was somewhat arbitrary) and released the changes to version 0.3.0 on the Marketplace. Thanks and let me know if you have any other issues.
